app/lib/package/name_tracker.dart

```diff
@@ -104,20 +104,20 @@ class NameTracker {
     }
   }
 
-  void addReservedName(String name) {
-    _reservedNames.add(name);
-    _addConflictingName(name, skipAlternatives: true);
-    _names.remove(name);
+  void addModeratedName(String name) {
+    _moderatedNames.add(name);
+    final existed = _names.remove(name);
+    if (existed) {
+      // reset conflicting names
+      _conflictingNames.removeWhere((key, value) => value == name);
+      for (final name in _names) {
+        _addConflictingName(name);
+      }
```

Comment:
Why is this necessary, are we cache state forever across runtimeVersions? that sounds dangerous.

Maybe we should fix the TODO, instead:
> /// TODO: support remove and re-scan package names every day or so.

Reply:
This is a cache for the local instance only, and runtimeVersions don't play much role in it.

We could re-scan daily, however, there are alternatives:
- This reset of the values when a ModeratedPackage first appears to the instance.
- With a bit more memory, we could track multiple source names for each conflicting alternative, and then removal would be much cheaper, and doesn't need re-calculation.

As package moderation is not a common event, I think we are better off with this, but I could also imagine daily re-scan or maybe a bit more memory use would be better in complexity.

Reply:
> As package moderation is not a common event, I think we are better off with this,

How is daily rescan not easier?

I just see a code path that rarely gets exercised.

Also we don't care if it takes 24 hours before a name can be reused...

Reply:
> How is daily rescan not easier?

It will need to have a separate data structure that we build on scanning, and some maintenance around triggering the scan. Not super-complex, but more lines of code than now..